pkg/git/git.go

```diff
@@ -46,119 +43,78 @@ var preReceiveHookTpl = template.Must(template.New("hooks").Parse(preReceiveHook
 
-	if ok := checkIfAllowed(repo, userinfo.Apps); !ok {
-		return nil, fmt.Sprintf("The user %v has no permission in application %v", userinfo.Username, repo)
+	if receivetype == "mock" {
+		channel.Write([]byte("OK"))
+		return nil
 	}
-
-	repo += ".git"
-
 	repoPath := filepath.Join(gitHome, repo)
-	log.Debugf(c, "creating repo directory %s", repoPath)
-	if _, err := createRepo(c, repoPath); err != nil {
+	log.Info("creating repo directory %s", repoPath)
+	if _, err := createRepo(repoPath); err != nil {
 		err = fmt.Errorf("Did not create new repo (%s)", err)
-		log.Warnf(c, err.Error())
-		return nil, err
+
+		return err
 	}
 
-	log.Debugf(c, "writing pre-receive hook under %s", repoPath)
-	if err := createPreReceiveHook(c, gitHome, repoPath); err != nil {
+	log.Info("writing pre-receive hook under %s", repoPath)
+	if err := createPreReceiveHook(gitHome, repoPath); err != nil {
 		err = fmt.Errorf("Did not write pre-receive hook (%s)", err)
-		log.Warnf(c, err.Error())
-		return nil, err
+		return err
 	}
 
 	cmd := exec.Command("git-shell", "-c", fmt.Sprintf("%s '%s'", operation, repo))
-	log.Infof(c, strings.Join(cmd.Args, " "))
+	log.Info(strings.Join(cmd.Args, " "))
 
 	var errbuff bytes.Buffer
 
 	cmd.Dir = gitHome
 	cmd.Env = []string{
-		fmt.Sprintf("RECEIVE_USER=%s", userinfo.Username),
+		fmt.Sprintf("RECEIVE_USER=%s", username),
 		fmt.Sprintf("RECEIVE_REPO=%s", repo),
-		fmt.Sprintf("RECEIVE_FINGERPRINT=%s", userinfo.Fingerprint),
+		fmt.Sprintf("RECEIVE_FINGERPRINT=%s", fingerprint),
 		fmt.Sprintf("SSH_ORIGINAL_COMMAND=%s '%s'", operation, repo),
-		fmt.Sprintf("SSH_CONNECTION=%s", c.Get("SSH_CONNECTION", "0 0 0 0").(string)),
+		fmt.Sprintf("SSH_CONNECTION=%s", conndata),
 	}
 	cmd.Env = append(cmd.Env, os.Environ()...)
 
-	log.Debugf(c, "Working Dir: %s", cmd.Dir)
-	log.Debugf(c, "Environment: %s", strings.Join(cmd.Env, ","))
+	log.Debug("Working Dir: %s", cmd.Dir)
+	log.Debug("Environment: %s", strings.Join(cmd.Env, ","))
 
 	inpipe, err := cmd.StdinPipe()
 	if err != nil {
-		return nil, err
+		return err
 	}
 	cmd.Stdout = channel
 	cmd.Stderr = io.MultiWriter(channel.Stderr(), &errbuff)
 
 	if err := cmd.Start(); err != nil {
 		err = fmt.Errorf("Failed to start git pre-receive hook: %s (%s)", err, errbuff.Bytes())
-		log.Warnf(c, err.Error())
-		return nil, err
+		return err
 	}
 
 	if _, err := io.Copy(inpipe, channel); err != nil {
 		err = fmt.Errorf("Failed to write git objects into the git pre-receive hook (%s)", err)
-		log.Warnf(c, err.Error())
-		return nil, err
+		return err
 	}
 
 	fmt.Println("Waiting for git-receive to run.")
 	fmt.Println("Waiting for deploy.")
 	if err := cmd.Wait(); err != nil {
 		err = fmt.Errorf("Failed to run git pre-receive hook: %s (%s)", errbuff.Bytes(), err)
-		log.Errf(c, err.Error())
-		return nil, err
+		return err
 	}
 	if errbuff.Len() > 0 {
-		log.Warnf(c, "Unreported error: %s", errbuff.Bytes())
+		log.Err("Unreported error: %s", errbuff.Bytes())
```

Comment:
should this return an error that represents what was in `errbuff`? may be outside of the scope of this PR